src/server/registry/catalog.go

```diff
@@ -75,7 +75,7 @@ func (r *repositoryHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 	}
 
 	// handle the pagination
-	resRepos := repoNames
+	var resRepos []string
```

Comment:
assuming `repoNames` is not empty
```
resRepos := repoNames
```
is not equivalent with
```
var resRepos []string
```

Reply:
It would be assigned value, so it is a ineffective assign here indeed.